tls/s2n_record_write.c

```diff
@@ -253,6 +253,106 @@ static inline int s2n_record_encrypt(
     return 0;
 }
 
+static S2N_RESULT s2n_record_write_mac(struct s2n_connection *conn, struct s2n_blob *record_header,
```

Comment:
Any interest in making this function more generic to suite the read case as well? Might be a massive refactor and not worth it, but we basically do the same thing on the read side when we need to manually calculate the mac.

Reply:
That's a good idea - I didn't think to reuse this on the read side. I started working on the refactor, but the change is getting big and it's touching a lot of sidetrail patches. I think it might still be worth doing, but it should probably go in its own PR.

Reply:
Issue for this: https://github.com/aws/s2n-tls/issues/4546

Reply:
Oh ouch, yeah, I don't want to block this PR on sidetrail patches.